### PR TITLE
[DOCS] Fingerprint ingest processor

### PR DESCRIPTION
**Pull request by danhermann**
Documentation for the processor added in #68415.

Depends on #68415.

### CODE REVIEW

File: docs/reference/ingest/processors/fingerprint.asciidoc [hunk 1]
```diff
@@ -0,0 +1,89 @@
+[role="xpack"]
+[testenv="basic"]
+[[fingerprint-processor]]
+=== Fingerprint processor
+++++
+<titleabbrev>Fingerprint</titleabbrev>
+++++
+
+Calculates a hash that can be used for content fingerprinting based on the
+content of the specified fields in a document.
```
Comment (jrodewig) [inline docs/reference/ingest/processors/fingerprint.asciidoc:10]: Nit: I'd split this into two sentences for clarity.
```suggestion
Computes a hash of the document's content. You can use this hash for
{wikipedia}/Fingerprint_(computing)[content fingerprinting].
```

File: docs/reference/ingest/processors/fingerprint.asciidoc [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+[[fingerprint-options]]
+.Fingerprint Options
+[options="header"]
+|======
+| Name               | Required | Default       | Description
+| `fields`           | yes      | n/a           | Array containing the names of the document
+fields whose content will be used to compute the fingerprint
```
Comment (jrodewig) [inline docs/reference/ingest/processors/fingerprint.asciidoc:18]: I'd clarify "content" here as we discussed offline. I'd also add a period for consistency.
```suggestion
| `fields`           | yes      | n/a           | Array of fields to include in
the fingerprint. For objects, the processor hashes both the field key and
value. For other fields, the processor hashes only the field value.
```

File: docs/reference/ingest/processors/fingerprint.asciidoc [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+| `target_field`     | no       | `fingerprint` | Output field for the fingerprint.
+| `salt`             | no       | <none>        | Salt value that will be applied to every
+fingerprint calculated by this processor
```
Comment (jrodewig) [inline docs/reference/ingest/processors/fingerprint.asciidoc:21]: Nit: I'd add a link and a period.
```suggestion
| `salt`             | no       | <none>        |
{wikipedia}/Salt_(cryptography)[Salt value] for the hash function.
```

File: docs/reference/ingest/processors/fingerprint.asciidoc [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+| `method`           | no       | `SHA-1`       | The cryptographic hash function used to
+compute the fingerprint. Must be one of `MD5`, `SHA-1`, `SHA-256`, or `SHA-512`.
+| `ignore_missing`   | no       | `false`       | If true, any missing elements in `fields`
+will be ignored when computing the fingerprint. If all fields are missing, the processor
+silently exits without modifying the document.
```
Comment (jrodewig) [inline docs/reference/ingest/processors/fingerprint.asciidoc:26]: Minor nit: I'd make `true` a literal and reword the first sentence for active voice.
```suggestion
| `ignore_missing`   | no       | `false`       | If `true`, the processor
ignores any missing `fields`. If all fields are missing, the processor silently
exits without modifying the document.
```

File: docs/reference/ingest/processors/fingerprint.asciidoc [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+include::common-options.asciidoc[]
+|======
+
+[discrete]
+[[fingerprint-processor-ex]]
+===== Examples
```
Comment (jrodewig) [inline docs/reference/ingest/processors/fingerprint.asciidoc:33]: Nit
```suggestion
===== Example
```

File: docs/reference/ingest/processors/fingerprint.asciidoc [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+            "is_active" : true
+          }
+        }
+      }
+    }
+  ]
+}
+----
+// TESTRESPONSE[s/\.\.\./"_index":"_index","_id":"_id","_ingest":{"timestamp":$body.docs.0.doc._ingest.timestamp},/]
+// NOTCONSOLE
```
Comment (jrodewig) [inline docs/reference/ingest/processors/fingerprint.asciidoc:89]: I tested this snippet locally with the changes in #68415. Should be good to remove `// NOTCONSOLE` once #68415 is merged.

**Review by jrodewig — APPROVED** [review]
LGTM.

I left a few minor nits but nothing blocking. Feel free to disregard if wanted. Thanks @danhermann!

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
Thanks, @jrodewig! I incorporated your suggestions in https://github.com/elastic/elasticsearch/pull/68610/commits/f865ba0aa85a4e2e57494a94085ee1dc57056382.

**Pull request merged**